tests/pipeline/test_python_component.py

```diff
@@ -141,18 +140,135 @@ def transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
             "name": "addn",
             "image": default_fondant_image,
             "description": "lightweight component",
-            "consumes": {"additionalProperties": True},
+            "consumes": {
+                "x": {"type": "int32"},
+                "y": {"type": "int32"},
+                "z": {"type": "int32"},
+            },
             "produces": {"additionalProperties": True},
             "args": {"n": {"type": "int"}},
         },
-        "consumes": {"x": {"type": "int32"}, "y": {"type": "int32"}},
-        "produces": {"x": {"type": "int32"}, "y": {"type": "int32"}},
+        "consumes": {},
+        "produces": {
+            "x": {"type": "int32"},
+            "y": {"type": "int32"},
+            "z": {"type": "int32"},
+        },
     }
     pipeline._validate_pipeline_definition(run_id="dummy-run-id")
 
     DockerCompiler().compile(pipeline)
 
 
+def test_consumes_mapping_all_fields(tmp_path_factory, load_pipeline):
```

Comment:
@RobbeSneyders this test is still failing. We are facing the same issue as discussed before when loading all the fields (consumed spec needs to be remapped based on the apply consumes before passing to the componentOp). Any other way of tackling this?

The rest of the cases (generic and specific consumes below) seem to work well

Reply:
I see two options:

- We accept the current behavior: you can't remap field names in the apply `consumes` if the component doesn't explicitly define `consumes` in the decorator.
- As you mention, include the remapping when calculating the default `consumes`. I think this is less of an issue than before, since this is no longer happening if the `consumes` is specified explicitly in the decorator.

Any other ideas?

If not, I would probably vote for the second one mentioned above.

Reply:
Indeed, the second option seems to be the most logical for now, will update it